packages/ui/components/form/selectimproved/components/Select.tsx

```diff
@@ -0,0 +1,259 @@
+import React, { useCallback, useEffect, useRef, useState } from "react";
+
+import { classNames as cn } from "@calcom/lib";
+import { useKeyPress } from "@calcom/lib/hooks/useKeyPress";
+
+import { FiX, FiChevronDown } from "../../../icon";
+import useOnClickOutside from "../hooks/use-onclick-outside";
+import Options from "./Options";
+import SearchInput from "./SearchInput";
+import SelectProvider from "./SelectProvider";
+import { Option } from "./type";
+
+interface SelectProps<T extends Option> {
+  options: T[];
+  selectedItems: T | T[] | null;
+  onChange: (value?: Option | Option[] | null) => void;
+  placeholder?: string;
+  isMultiple?: boolean;
+  isClearable?: boolean;
+  isSearchable?: boolean;
+  isDisabled?: boolean;
+  loading?: boolean;
+  menuIsOpen?: boolean;
+  searchInputPlaceholder?: string;
+  noOptionsMessage?: string;
+  classNames?: {
+    menuButton?: ({ isDisabled }: { isDisabled: boolean }) => string;
+    menu?: string;
+    tagItem?: ({ isDisabled }: { isDisabled: boolean }) => string;
+    tagItemText?: string;
+    tagItemIconContainer?: string;
+    tagItemIcon?: string;
+    list?: string;
+    listGroupLabel?: string;
+    listItem?: ({ isSelected }: { isSelected: boolean }) => string;
+    listDisabledItem?: string;
+    ChevronIcon?: ({ open }: { open: boolean }) => string;
+    searchContainer?: string;
+    searchBox?: string;
+    searchIcon?: string;
+    closeIcon?: string;
+  };
```

Comment:
WIP styles override if needed

Reply:
Nice — I think this is a super simple approach of overriding the styles. Very easy to use, clear on how it works 👏